test/test_unstructured_2d.jl

```diff
@@ -323,6 +323,33 @@ end
     end
 end
 
+@trixi_testset "elixir_shallowwater_ec_float32.jl" begin
+    # Expected errors are nearly all taken from elixir_shallowwater_ec.jl
+    @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_shallowwater_ec_float32.jl"),
+                        l2=[
+                            Float32(0.6107326269462766),
+                            Float32(0.48666631722018877),
+                            Float32(0.48309775159067053),
+                            Float32(0.29467422718511704),
+                        ],
+                        linf=[
+                            Float32(2.776782342826098),
+                            3.2162943f0, # this needs to be adapted
+                            3.6683278f0, # this needed to be adapted
+                            Float32(2.052861364219655),
```

Comment:
Ah, this answers my question above. It is odd that the `l2` error values did not need altered. In the `UnstructuredMesh2D` constructor do we need to have `convert` calls for terms like the node element corners or the boundary polynomial data because the mesh file provides this information in double precision?

Reply:
The `UnstructuredMesh2D` has a type parameter `RealT` that allows us to do exactly this conversion 👍

Reply:
Gotcha, that makes portability of code quite slick.